app/views/layouts/_top_bar.html.erb

```diff
@@ -50,8 +50,9 @@
       }
 
       // Load notifications count as quickly as it's available.
+      // Not if we're on the notifications page itself
       if (
-        document.getElementById('notifications-container') == null
+        window.location.pathname !== '/notifications'
       ) {
         var xmlhttp;
         if (window.XMLHttpRequest) {
```

Comment:
A couple of suggestions which could be in another small PR is just some cleanup, e.g. `document.getElementById('notifications-number')` could be defined once before the if/else and consider moving towards `fetch`.

Reply:
Yeah, definitely